ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,668 @@
+  city: string;
+  state: string;
+  zipcode: string;
+  country: string;
+}
+
+export interface AccountCreationProps {
+  invitationKey: string;
+  setProfile: Function;
+}
+
+export interface AccountCreationState {
+  creatingAccount: boolean;
+  errors: any;
+  invalidEmail: boolean;
+  rolesOptions: any;
+  profile: Profile;
+  showAllFieldsRequiredError: boolean;
+  showInstitution: boolean;
+  showAffiliationRole: boolean;
+  showAffiliationOther: boolean;
+  showNext: boolean;
+  usernameCheckInProgress: boolean;
+  usernameConflictError: boolean;
+  address: Address;
+  institutionName: string;
+  institutionRole: string;
+  affiliation: string;
+  affiliationRole: string;
+  affiliationOther: string;
+}
+
+const styles = {
+  sectionLabel: {
+    height: '22px',
+    color: colors.primary,
+    fontFamily: 'Montserrat',
+    fontSize: '16px',
+    fontWeight: 600,
+    lineHeight: '22px',
+    paddingBottom: '1.5rem'
+  },
+  section: {
+    width: '12rem'
+  }
+};
+
+const nameLength = 80;
+
+export const Section = (props) => {
+  return <FormSection
+      style={{display: 'flex', flexDirection: 'column', paddingTop: '1rem', ...props.style}}>
+    <label style={styles.sectionLabel}>
+      {props.header}
+    </label>
+    {props.children}
+  </FormSection>;
+};
+
+export class AccountCreation extends React.Component<AccountCreationProps, AccountCreationState> {
+  private usernameCheckTimeout: NodeJS.Timer;
+
+  constructor(props: AccountCreationProps) {
+    super(props);
+    this.state = {
+      errors: undefined,
+      profile: {
+        // Note: We abuse the "username" field here by omitting "@domain.org". After
+        // profile creation, this field is populated with the full email address.
+        username: '',
+        dataAccessLevel: DataAccessLevel.Unregistered,
+        givenName: '',
+        familyName: '',
+        contactEmail: '',
+        currentPosition: '',
+        organization: '',
+        areaOfResearch: '',
+      },
+      usernameCheckInProgress: false,
+      usernameConflictError: false,
+      creatingAccount: false,
+      showAllFieldsRequiredError: false,
+      showInstitution: true,
+      showAffiliationRole: false,
+      showAffiliationOther: false,
+      showNext: false,
+      invalidEmail: false,
+      rolesOptions: [],
+      address: {
+        streetAddress1: '',
+        streetAddress2: '',
+        city: '',
+        state: '',
+        zipcode: '',
+        country: ''
+      },
+      institutionName: '',
+      institutionRole: '',
+      affiliation: '',
+      affiliationRole: '',
+      affiliationOther: ''
+    };
+  }
+
+  // This method will be modified once the story  below will be done
+  // https://precisionmedicineinitiative.atlassian.net/browse/RW-3284
+  createAccount(): void {
+    const {invitationKey, setProfile} = this.props;
+    const profile = this.state.profile;
+    const emailValidRegex = new RegExp(/^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/);
```

Comment:
Please use a library to validate email addresses. There are tons of edge cases. For example, it looks like this won't support the plus character, which gmail uses for lightweight aliases.

We need lots of unit tests too.